### PR TITLE
fat(pipelines/pingcap/tidb): add report for `ghpr_build`

### PR DESCRIPTION
**Pull request by wuhuizuo**
replay link: https://prow-ee.pingcap.net/jenkins/job/pingcap/job/tidb/job/ghpr_build/1459/console

new:
- [x] extract report shell steps to a shell script.
- [ ] todo: support support JUnit xml url setting args(should be next PR when migrate other test pipeline)

### CODE REVIEW

File: pipelines/pingcap/tidb/latest/ghpr_build.groovy [hunk 1]
```diff
@@ -124,7 +124,6 @@ pipeline {
             }
         }
         stage("Build tidb-server and plugin"){
-            failFast true
```
Comment (wuhuizuo) [inline pipelines/pingcap/tidb/latest/ghpr_build.groovy:127]: why: enabled in pipeline top level.

File: pipelines/pingcap/tidb/latest/ghpr_build.groovy [hunk 2]
```diff
@@ -206,4 +202,66 @@ pipeline {
...
+                // 4 : 'flaky test error',
+                // 5 : 'fmt or lint fail',
+                // 6 : 'pipeline engine error',
+                // 7 : 'agent env error',
+                // 8 : 'other error',
+                // ============================================================                
+                def ciErrorCode = currentBuild.result == "SUCCESS" ? 0 : 2
+                def resultDownloadPath = "${FILE_SERVER_URL}/download/builds/pingcap/tidb-check/pr/${ghprbActualCommit}/centos7/tidb-server.tar.gz"
+                sh """
+                taskStartTime="\$(stat /proc/1 | grep 'Modify: ' | sed 's/Modify: //')"
```
Comment (purelind) [inline pipelines/pingcap/tidb/latest/ghpr_build.groovy:224]: Is this the pipeline start time in millisecond?

Comment (wuhuizuo) [reply pipelines/pingcap/tidb/latest/ghpr_build.groovy:224]: > Is this the pipeline start time in millisecond?
next line will transfer to unix milliseconds

**Review by purelind — APPROVED** [review]
LGTM

**Pull request merged**